### PR TITLE
feat(UVE): Enable Editing Pages with Custom Query

### LINKED ISSUE #29263: UVE: Enable Editing Pages with Custom Query
### User Story
As a content editor, I need the capability to edit pages generated from various request types or queries to dotCMS data through the Universal Visual Editor (UVE). Some queries could be:

1. GraphQL-specific query or multiples queries
2. Page API with custom parameters like depth, etc.

### The Problem
The problem is that we can't know what request the developer did to generated pages, **we don't know exactly what properties** the user requested to render their page or which GQL query.

When the page loads inside the editor, we send the page data to render that page, and the data we send is coming from the Page API which will often defer from the data that the customer requests in GQL.

The solution is **to receive the GQL query (or `client.page.get` params)** that the customer used to render the page and use that to get the page data we'll be sending down to the client while inside UVE.

We need to update our SDK, libraries and documentation so we can register that “request info”.

### Things to consider
1. User can use any GQL client library
3. User can use complex GQL queries with fragments
4. User can use [inline fragments](https://graphql.org/learn/queries/#inline-fragments), not the same as fragments
5. User can have queries with names
6. User can have multiples queries in one request.

```[tasklist]
### Acceptance Criteria
- [x] UVE should be updated to support editing pages created with GraphQL data.
- [x] Documentation on how to implement GraphQL pages should be updated to include instructions on editing those pages using UVE.
```

### Proposed Objective
Core Features

### Proposed Priority
Priority 2 - High

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
N/A

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
N/A

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Reduce the number of initial load requests from 9 to 3.
   The remaining 3 necessary requests are: `Guard`, `Page`, `Iframe` [Client]

### Checklist
- [x] Tests

#### NextJS with GraphQL

https://github.com/user-attachments/assets/ad5086da-399b-45b5-ac26-188e19a292ea

#### NextJS Page with `depth=2`

https://github.com/user-attachments/assets/fdfa5bf4-4b6e-444e-be92-4431d956b757

#### Angular Page with `depth=2`

https://github.com/user-attachments/assets/85603460-c417-4a38-ba2b-d016c9311e28

#### Traditional Page (VTL)

https://github.com/user-attachments/assets/7e7b0b7f-14b6-44fa-8615-43a953cf5ff4

This PR fixes: #29263

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/client/withClient.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                    return {
+                        query,
+                        params: {
+                            ...store.params(),
+                            ...params
+                        }
+                    };
+                })
+            };
+        }),
```
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/client/withClient.ts]: Computed props should be a reflection of the UI and here you are just accessing the state. Do we need this computed? I think we can do the same by just accessing the store directly.

In the place we call the PageAPI we already have the params, we can do the merge there. I think we should separate the params of the editor from the params the user sends from the client.

See comments below.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/save/withSave.ts [hunk 2]
```diff
@@ -40,10 +42,11 @@ export function withSave() {
                                 pageId: store.pageAPIResponse().page.identifier,
                                 params: store.params()
                             };
+                            const clientProps = store.$clientRequestProps();
```
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/save/withSave.ts:45]: Above this line we have a payload that already have the params, maybe we should adapt this payload to work also with the Client Params and do the merge logic inside the pageAPI method

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/withEditor.ts [hunk 3]
```diff
@@ -85,9 +88,11 @@ export function withEditor() {
...
                     return {
                         code: store.pageAPIResponse()?.page?.rendered,
-                        isTraditionalPage: store.isTraditionalPage(),
-                        isEditState: store.isEditState(),
-                        isEnterprise: store.isEnterprise()
+                        isClientReady: store.isClientReady(),
+                        isTraditionalPage: untracked(() => store.isTraditionalPage()),
+                        enableInlineEdit: untracked(
+                            () => store.isEditState() && store.isEnterprise()
+                        )
```
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/withEditor.ts:95]: This should be tracked, `isEditState` changes when we select `device` or `socialMedia`

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/client/withClient.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/client/withClient.ts]: That makes sense, that was my first approach but we could use the computed for methods unrelated to the UI

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/withEditor.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/withEditor.ts:95]: Done!

**Review by zJaaal — APPROVED** [review]

File: examples/angular/src/app/pages/pages.component.ts [hunk 4]
```diff
@@ -41,17 +41,15 @@ export class DotCMSPagesComponent implements OnInit, OnDestroy {
...
+
   ngOnInit() {
     // Get the context data from the route
     this.route.data
       .pipe(takeUntilDestroyed(this.destroyRef))
       .subscribe((data) => {
         this.context.set(data['context']);
       });
-
-    this.client.editor.on('changes', (pageAsset) => {
```
Comment (KevinDavilaDotCMS) [inline examples/angular/src/app/pages/pages.component.ts:52]: Is this not longer necessary?
I see this was moved to useDotcmsEditor, but the changes can be reflected at the "pages" (client application) level?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/save/withSave.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/save/withSave.ts]: Done!

File: examples/angular/src/app/pages/pages.component.ts (hunk 4 above)
Comment (rjvelazco) [reply examples/angular/src/app/pages/pages.component.ts:52]: That's right.

Now, we are listening to those changes from the `useDotcmsEditor`. Every time a change is made in the UVE, it gets reflected at the user level.

Comment (rjvelazco) [reply examples/angular/src/app/pages/pages.component.ts:52]: Same with Angular!

Comment (rjvelazco) [reply examples/angular/src/app/pages/pages.component.ts:52]: @KevinDavilaDotCMS 

https://github.com/user-attachments/assets/f2e9be3f-9f1c-472f-86d6-72bce1c5ac1d

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Pull request merged**